blocks/section-header/section-header.css

```diff
@@ -36,5 +36,28 @@ main > .section > .section-header {
   .section-header__description + .section-header__button {
     margin-inline-start: auto;
     grid-column: 9 / span 4;
+    grid-row: 1 / span 2;
+    align-self: end;
+  }
+}
+
+.grid-item--66 .section-header,
+.grid-item--50 .section-header {
+  /* styles for when a section header with button is inside a limited grid container */
+
+  @media (min-width: 48rem) {
+    .section-header__heading.grid-item--66,
+    .section-header__description.grid-item--66 {
+      grid-column: span 12;
+      max-width: var(--page-constrained-text);
+    }
+
+    .section-header__description + .section-header__button {
+      margin-inline-start: unset;
+      grid-column: span 12;
+      grid-row: 3;
+      align-self: unset;
+      margin-block-start: 1rem;
+    }
```

Comment:
this is an interesting one. instead of using container queries, i look for the grid classes on the containers and overwrote the relevant styles